adafruit_ads1x15/ads1x15.py

```diff
@@ -69,66 +71,70 @@ def __init__(
         i2c: I2C,
         gain: float = 1,
         data_rate: Optional[int] = None,
-        mode: Mode = Mode.SINGLE,
+        mode: int = Mode.SINGLE,
         address: int = _ADS1X15_DEFAULT_ADDRESS,
     ):
         # pylint: disable=too-many-arguments
         self._last_pin_read = None
         self.buf = bytearray(3)
-        self._data_rate = self._gain = self._mode = None
         self.gain = gain
         self.data_rate = self._data_rate_default() if data_rate is None else data_rate
         self.mode = mode
         self.i2c_device = I2CDevice(i2c, address)
 
     @property
-    def data_rate(self):
+    def bits(self) -> Union[Literal[12], Literal[16]]:
```

Comment:
Since, you're saying the subclass should implement this, let's type it as an `int` return to say that we're promising that bits will return an integer, and we can be more specific as we have in said subclasses.